flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecGroupWindowAggregate.java

```diff
@@ -0,0 +1,373 @@
+            } else if (isRowtimeAttribute(timeField) && hasTimeIntervalType(size)) {
+                builder = builder.tumble(toDuration(size)).withEventTime(timeFieldIndex);
+            } else if (isProctimeAttribute(timeField) && hasRowIntervalType(size)) {
+                builder = builder.countWindow(toLong(size));
+            } else {
+                // TODO: EventTimeTumblingGroupWindow should sort the stream on event time
+                // before applying the  windowing logic. Otherwise, this would be the same as a
+                // ProcessingTimeTumblingGroupWindow
+                throw new UnsupportedOperationException(
+                        "Event-time grouping windows on row intervals are currently not supported.");
+            }
+        } else if (window instanceof SlidingGroupWindow) {
+            SlidingGroupWindow slidingWindow = (SlidingGroupWindow) window;
+            FieldReferenceExpression timeField = slidingWindow.timeField();
+            ValueLiteralExpression size = slidingWindow.size();
+            ValueLiteralExpression slide = slidingWindow.slide();
+            if (isProctimeAttribute(timeField) && hasTimeIntervalType(size)) {
+                builder = builder.sliding(toDuration(size), toDuration(slide)).withProcessingTime();
+            } else if (isRowtimeAttribute(timeField) && hasTimeIntervalType(size)) {
+                builder =
+                        builder.sliding(toDuration(size), toDuration(slide))
+                                .withEventTime(timeFieldIndex);
+            } else if (isProctimeAttribute(timeField) && hasRowIntervalType(size)) {
+                builder = builder.countWindow(toLong(size), toLong(slide));
+            } else {
+                // TODO: EventTimeTumblingGroupWindow should sort the stream on event time
+                // before applying the  windowing logic. Otherwise, this would be the same as a
+                // ProcessingTimeTumblingGroupWindow
+                throw new UnsupportedOperationException(
+                        "Event-time grouping windows on row intervals are currently not supported.");
+            }
+        } else if (window instanceof SessionGroupWindow) {
+            SessionGroupWindow sessionWindow = (SessionGroupWindow) window;
+            FieldReferenceExpression timeField = sessionWindow.timeField();
+            ValueLiteralExpression gap = sessionWindow.gap();
+            if (isProctimeAttribute(timeField)) {
+                builder = builder.session(toDuration(gap)).withProcessingTime();
+            } else if (isRowtimeAttribute(timeField)) {
+                builder = builder.session(toDuration(gap)).withEventTime(timeFieldIndex);
+            } else {
+                throw new UnsupportedOperationException("This should not happen.");
+            }
+        } else {
+            throw new TableException("Unsupported window: " + window.toString());
+        }
+
+        if (emitStrategy.produceUpdates()) {
```

Comment:
The original implementation of `StreamExecGroupWindowTableAggregate` throws exception if `emitStrategy` produces updates. However, this check is removed now? 

Maybe we can throw this exception in `WindowOperatorBuilder` to keep this class unchanged.

Reply:
No, the code snippet is in `StreamPhysicalGroupWindowTableAggregate`